File: backend/cmd/models/Studio.go
package models

import "gorm.io/gorm"

type SimpleStudio struct {
	ID      uint
	Name    string
	Thumb   string
	Country string
}

type Studio struct {
	gorm.Model
	Name    string   `gorm:"not null;index"`
	Thumb   string   `gorm:"not null;default:'no_thumb.png'"`
	Country string   `gorm:"not null;default:'unknown'"`
	Movies  []*Movie `gorm:"many2many:movie_studios" json:"movies"`
}

File: backend/models/Studio.js
import mongoose from "mongoose";

const StudioSchema = new mongoose.Schema({
  name: {
    type: String,
    required: [true, "name is required"],
    maxLength: [80, "name must not be longer than 80 characters"],
    trim: true,
  },

  country: {
    type: String,
    default: "unknown",
    trim: true,
    maxLength: [20, "country must not be longer than 20 characters"],
  },
});

const Studio = mongoose.model("studio", StudioSchema);

export default Studio;
